web-module/src/main/webapp/resources/structEditJS/StructEditor.mjs

```diff
@@ -0,0 +1,878 @@
+              console.debug('finishLoadingTinymMce: skip event for ', editor.id, ' waiting for ',
+                formfield.id);
+            }
+          };
+          $$('body')[0].observe('celRTE:finishedInit', finishLoadingTinymMce);
+        });
+      }
+    } else {
+      console.debug('_startGetFieldValueAsync: no tinyMCE or no mceEditor field ', formfield.id);
+    }
+    return null;
+  },
+
+  _getFieldValue : function(formfield, cblFunc) {
+    const _me = this;
+    console.debug('_startGetFieldValueAsync: _startGetFieldValueAsync for ', formfield.id);
+    const possiblePromise = _me._startGetFieldValueAsync(formfield, cblFunc);
+    if (possiblePromise) {
+      possiblePromise
+      .then(cblFunc)
+      .catch(exp => {
+        console.error('_getFieldValue: failed with exception ' + formfield.id, exp);
+      });
+    } else {
+      cblFunc(formfield.value);
+    }
+  },
+
+  _updateTinyMCETextAreas : function() {
+    const _me = this;
+    const formId = _me.getFormId();
+    const mceFields = document.forms[formId].select('textarea.mceEditor');
+    if (typeof tinyMCE !== 'undefined') {
+      console.log('_updateTinyMCETextAreas: for ', formId, mceFields);
+      mceFields.each(function(formfield) {
+        _me._updateOneTinyMCETextArea(formfield);
+      });
+    } else {
+      console.debug('_updateTinyMCETextAreas: skip no tinyMCE');
+    }
+    console.log('_updateTinyMCETextAreas: end ', formId);
+  },
+
+  /**
+   * submittable fields must have a name attribute and maynot be disabled
+   * @param fieldElem
+   * @returns {Boolean}
+   */
+  _isSubmittableField : function(fieldElem) {
+    return (fieldElem.name && (fieldElem.name != '') && !fieldElem.disabled);
+  },
+
+  _retrieveInitialValues : function() {
+    const formId = this.getFormId();
+    if (this.isValidFormId()) {
+      const formdata = new FormData(document.getElementById(formId));
+      const newInitialValues = {};
+      [...formdata.keys()].forEach(key => {
+        newInitialValues[key] = new Set(formdata.getAll(key));
+      });
+      this._initialValues = Object.freeze(newInitialValues);
+      console.log('retrieveInitialValues: ', formId, this._initialValues);
+    }
+  },
+
+  _formDirtyOnLoad : function() {
+    var _me = this;
+    return  (typeof(_me._formElem.celFormDirtyOnLoad) !== 'undefined')
+        && (_me._formElem.celFormDirtyOnLoad.value == 'true');
+  },
+
+  /**
+   * _isDirtyField and needs saving
+   *
+   * @param fieldElem
+   * @return
+   */
+  _isDirtyField : function(fieldElem) {
+    if (fieldElem.hasClassName('celDirtyOnLoad')) {
+      return true;
+    }
+    const formElem = fieldElem.closest('form');
+    const formdata = new FormData(formElem);
+    const formElements = formElem.elements;
+    const dirtyFields = [...formdata.keys()].filter(key => 
+      !formElements[key].classList.contains('celIgnoreDirty')
+      && !this._equalsParamValues(formdata.getAll(key), this._initialValues[key]));
```

Comment:
I guess FormData keys needn't necessarily exist in `_initialValues`:
```suggestion
      && !this._equalsParamValues(formdata.getAll(key), this._initialValues[key] ?? new Set()));
```

Reply:
fixed in L755 (function definition)